rust/connlib/clients/shared/Cargo.toml

```diff
@@ -29,6 +29,8 @@ tokio-tungstenite = { version = "0.21", default-features = false, features = ["c
 async-compression = { version = "0.4.3", features = ["tokio", "gzip"] }
 hickory-resolver = { workspace = true, features = ["tokio-runtime"] }
 parking_lot = "0.12"
+bimap = "0.6"
+ip_network = { version = "0.4", default-features = false }
```

Comment:
I see this `ip_network` dep also mentioned in the Cargo.toml for individual clients, and for the tunnel, though sometimes with different features. Does that mean clients will get multiple copies, or it'll merge the feature sets somehow? Would it be easier to put it up in the workspace Cargo.toml?

Reply:
AFAIK the dependency will only be linked once as long as it can be, since feature flags should be additive it will also merege feature sets.

I prefer to do it like this unless there's an actual requirement for the crate to be the same version(such as sharing types/traits) since that means we can update the dependency on a semver breaking manner for some crates but not for others where it is more work.